Source/ResponseSerialization.swift

```diff
@@ -64,18 +64,25 @@ public protocol DownloadResponseSerializerProtocol {
 
 /// A serializer that can handle both data and download responses.
 public protocol ResponseSerializer: DataResponseSerializerProtocol & DownloadResponseSerializerProtocol {
+    /// Closure type used to prepare `Data` for serialization.
+    typealias DataPreprocessor = (Data) throws -> Data
```

Comment:
I really like this change overall. It really gives you a lot of flexibility if you need it. I think for users that actually need this functionality though might be better served by this being a protocol rather than a closure. I think most people would want to define an explicit type to very easily test rather than having to implement it as a closure. I just think it could be some fairly complex logic that people may want to reuse more easily. They could pass functions off as the closure, but that doesn't seem ideal.

In the XSSI issue, it seems like it would be nice to be able to create an `XSSIDataPreprocessor: DataPreprocessor` that you could extensively unit test and leverage throughout your code. So I think I lean towards having `DataPreprocessor` be a protocol rather than a typealias'd closure. Either way, both solutions are solid.

Reply:
I updated to a protocol in https://github.com/Alamofire/Alamofire/pull/2903/commits/6d1a440ceb7d3bd8ae6aed4cc923d17bf202c0fa.